app/src/main/java/com/example/pokemonapp/fragment/CardTypeFragment.kt

```diff
@@ -5,14 +5,54 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.GridLayoutManager
+import com.example.pokemonapp.adapter.TypeAdapter
 import com.example.pokemonapp.databinding.FragmentCardTypeBinding
+import com.example.pokemonapp.utils.Event
+import com.example.pokemonapp.viewmodel.CardTypeViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class CardTypeFragment : Fragment() {
 
+    private lateinit var adapter: TypeAdapter
     private lateinit var binding: FragmentCardTypeBinding
+    private val viewModel by viewModel<CardTypeViewModel>()
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
         binding = FragmentCardTypeBinding.inflate(inflater, container, false)
+        viewModel.getLiveDataState().observe(viewLifecycleOwner, updateUIObserver)
+        viewModel.getTypes()
+        initializeRecyclerView()
         return binding.root
     }
+
+    private val updateUIObserver = Observer<Event<CardTypeViewModel.CardTypeData>> { event ->
+        val eventContent = event.getContentIfNotHandled()
+        when (eventContent?.responseType) {
+            CardTypeViewModel.CardTypeStatus.SUCCESSFUL -> {
+                binding.progressBarCardTypeFragmentLoad.visibility = View.GONE
+                eventContent.data.data.let {
+                    adapter.setTypes(it)
+                }
+            }
+            CardTypeViewModel.CardTypeStatus.ERROR -> {
+                binding.progressBarCardTypeFragmentLoad.visibility = View.GONE
+                binding.textViewCardTypeFragmentTypesError.text = eventContent.error
+            }
+            CardTypeViewModel.CardTypeStatus.LOADING -> {
+                binding.progressBarCardTypeFragmentLoad.visibility = View.VISIBLE
+            }
+        }
+    }
+
+    private fun initializeRecyclerView() {
+        binding.recyclerViewCardTypeFragmentTypesList.layoutManager = GridLayoutManager(context, COLUMS_AMOUNT)
```

Comment:
here you can do something like this 
```
binding.reclyclerviewCardTypeFragmment.apply {
    layoutManager = GridLayoutManager(context, COLUMS_AMOUNT)    
    adapter = TypeAdapter(emptyList())
}
```
but why you first gettypes and then you initialize the adapter